internal/devices/update.go

```diff
@@ -51,48 +53,59 @@ func (c *Client) Update() *cobra.Command {
 
 		RunE: func(cmd *cobra.Command, args []string) error {
 			cmd.SilenceUsage = true
-			req := &packngo.DeviceUpdateRequest{}
+			deviceUpdate := metal.NewDeviceUpdateInput()
 
 			if hostname != "" {
-				req.Hostname = &hostname
+				deviceUpdate.Hostname = &hostname
 			}
 
 			if description != "" {
-				req.Description = &description
+				deviceUpdate.Description = &description
 			}
 
 			if userdata != "" {
-				req.UserData = &userdata
+				deviceUpdate.Userdata = &userdata
 			}
 
 			if locked {
-				req.Locked = &locked
+				deviceUpdate.Locked = &locked
 			}
 
 			if len(tags) > 0 {
-				req.Tags = &tags
+				deviceUpdate.Tags = tags
+			}
+
+			if f := cmd.Flag("always-pxe"); f.Changed {
+				deviceUpdate.SetAlwaysPxe(true)
```

Comment:
I think this will cause problems for devices that don't use custom iPXE; we should only set `alwaysPxe` on the device request if the `always-pxe` flag was explicitly set to either `true` or `false`.